src/Symfony/Component/Messenger/Middleware/SendMessageMiddleware.php

```diff
@@ -107,4 +103,18 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
         // message should only be sent and not be handled by the next middleware
         return $envelope;
     }
+
+    /**
+     * * @return iterable|SenderInterface[]
+     */
+    private function getSenders(Envelope $envelope, &$handle, ?RedeliveryStamp $redeliveryStamp): iterable
+    {
+        if (null !== $redeliveryStamp) {
+            return [
+                $redeliveryStamp->getSenderClassOrAlias() => $this->sendersLocator->getSenderByAlias($redeliveryStamp->getSenderClassOrAlias()),
```

Comment:
For redelivery, the sender "alias" really becomes more important. This is communicated on the phpdoc of `SendersLocatorInterface::getSenders()`, which mentions that the iterator should be keyed by a sender alias, when possible.